monty-31/src/dft/backward.rs

```diff
@@ -108,40 +281,24 @@ impl<MP: MontyParameters + TwoAdicData> MontyField31<MP> {
         Self::backward_pass(a, &root_table[0]);
     }
 
-    #[inline(always)]
-    fn backward_64(a: &mut [Self], root_table: &[Vec<Self>]) {
-        assert_eq!(a.len(), 64);
-
-        // Safe because a.len() == 64
-        let (a0, a1) = unsafe { a.split_at_mut_unchecked(a.len() / 2) };
-        Self::backward_32(a0, &root_table[1..]);
-        Self::backward_32(a1, &root_table[1..]);
-
-        Self::backward_pass(a, &root_table[0]);
-    }
-
-    #[inline(always)]
-    fn backward_128(a: &mut [Self], root_table: &[Vec<Self>]) {
-        assert_eq!(a.len(), 128);
-
-        // Safe because a.len() == 128
-        let (a0, a1) = unsafe { a.split_at_mut_unchecked(a.len() / 2) };
-        Self::backward_64(a0, &root_table[1..]);
-        Self::backward_64(a1, &root_table[1..]);
-
-        Self::backward_pass(a, &root_table[0]);
-    }
+    /// Assumes `a.len() > 8`
+    #[inline]
+    fn backward_fft_recur(input: &mut [Self], root_table: &[Vec<Self>]) {
```

Comment:
For the doc comment, is `a` input? Also I think the assumption should be `input.len() > 64`.